File: 1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.cpp
class Solution {
public:
    int minimumDeletions(string s) {
        
        int count_right_a = 0;

        for(int i=s.size()-1;i>=0;i--){
            if(s[i]=='a') count_right_a++;
        }

        int count_b = 0;

    int min_deletions = INT_MAX;

        for(int i=0;i<s.size();i++){
        
            if(s[i]=='a') count_right_a--;

            min_deletions= min(min_deletions, count_b + count_right_a);

            if(s[i]=='b') count_b++;

        }

    return min_deletions;

    }
};

File: 1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.java
class Solution {
    public int minimumDeletions(String s) {
        int count_right_a = 0;

        for(int i=s.length()-1; i>=0;i--){
            if(s.charAt(i)=='a') count_right_a++;
        }

        int count_b=0;
        int min_deletions = Integer.MAX_VALUE;

        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='a') count_right_a--;
            
            min_deletions = Math.min(min_deletions, count_b + count_right_a);

            if(s.charAt(i)=='b') count_b++;
        }
    return min_deletions;
    }
}